engine/src/main/java/org/terasology/engine/modes/loadProcesses/CreateWorldEntity.java

```diff
@@ -52,64 +55,101 @@ public String getMessage() {
-
-            // transfer all world generation parameters from Config to WorldEntity
-            WorldGenerator worldGenerator = context.get(WorldGenerator.class);
-            SimpleUri generatorUri = worldGenerator.getUri();
-            Config config = context.get(Config.class);
-
-            // get the map of properties from the world generator.
-            // Replace its values with values from the config set by the UI.
-            // Also set all the components to the world entity.
-            WorldConfigurator worldConfigurator = worldGenerator.getConfigurator();
-            Map<String, Component> params = worldConfigurator.getProperties();
-            for (Map.Entry<String, Component> entry : params.entrySet()) {
-                Class<? extends Component> clazz = entry.getValue().getClass();
-                Component comp = config.getModuleConfig(generatorUri, entry.getKey(), clazz);
-                if (comp != null) {
-                    worldEntity.addComponent(comp);
-                    worldConfigurator.setProperty(entry.getKey(), comp);
-                } else {
-                    worldEntity.addComponent(entry.getValue());
+        };
+        setProperties(context.get(WorldConfigurator.class), configuratorConfigurator);
+    }
+
+    /** transfer all world generation parameters from Config to WorldEntity */
+    private WorldConfigurator useConfigurationFromConfig() {
+        WorldConfigurator worldConfigurator = context.get(WorldConfigurator.class);
+        SimpleUri generatorUri = context.get(WorldGenerator.class).getUri();
+        Config config = context.get(Config.class);
+        setProperties(worldConfigurator, new WorldConfiguratorConfigurator() {
+                    @Override
+                    public <T extends Component> T getComponentOfWorldEntity(String key, Class<T> clazz) {
+                        return config.getModuleConfig(generatorUri, key, clazz);
+                    }
                 }
+        );
+        return worldConfigurator;
+    }
+
+    private EntityRef getExistingWorldEntity() {
+        EntityManager entityManager = context.get(EntityManager.class);
+        Iterator<EntityRef> worldEntityIterator = entityManager.getEntitiesWith(WorldComponent.class).iterator();
+        EntityRef worldEntity = worldEntityIterator.next();
+        worldEntityIterator.forEachRemaining(w -> logger.warn("Ignored extra world {}", w));
+        return worldEntity;
+    }
+
+    private EntityRef createWorldPoolsAndEntity() {
+        EntityManager entityManager = context.get(EntityManager.class);
+
+        entityManager.createWorldPools(gameManifest);
+
+        EntityRef worldEntity = entityManager.create();
+        worldEntity.addComponent(new WorldComponent());
+        NetworkComponent networkComponent = new NetworkComponent();
+        networkComponent.replicateMode = NetworkComponent.ReplicateMode.ALWAYS;
+        worldEntity.addComponent(networkComponent);
```

Comment:
I was wondering if this should use a Component Builder, but it didn't seem to make this easier to read and I didn't need to otherwise mess with this block, so I left it alone.